src/IO/H5/EosTable.cpp

```diff
@@ -0,0 +1,120 @@
+    std::vector<std::array<double, 2>> independent_variable_bounds,
+    std::vector<size_t> independent_variable_number_of_points,
+    std::vector<bool> independent_variable_uses_log_spacing,
+    const bool beta_equilibrium, const uint32_t version)
+    : group_(std::move(group)),
+      name_(name.size() > extension().size()
+                ? (extension() == name.substr(name.size() - extension().size())
+                       ? name
+                       : name + extension())
+                : name + extension()),
+      path_(group_.group_path_with_trailing_slash() + name),
+      version_(version),
+      eos_table_group_(group_.id(), name_, h5::AccessType::ReadWrite),
+      independent_variable_names_(std::move(independent_variable_names)),
+      independent_variable_bounds_(std::move(independent_variable_bounds)),
+      independent_variable_number_of_points_(
+          std::move(independent_variable_number_of_points)),
+      independent_variable_uses_log_spacing_(
+          std::move(independent_variable_uses_log_spacing)),
+      beta_equilibrium_(beta_equilibrium) {
+  if (subfile_exists) {
+    ERROR(
+        "Opening an equation of state table with the constructor for writing a "
+        "table, but the subfile "
+        << name << " already exists");
+  }
+  // Subfiles are closed as they go out of scope, so we have the extra
+  // braces here to add the necessary scope
+  {
+    Version open_version(false, detail::OpenGroup{}, eos_table_group_.id(),
+                         "version", version_);
+  }
+  {
+    Header header(false, detail::OpenGroup{}, eos_table_group_.id(), "header");
+    header_ = header.get_header();
+  }
+
+  h5::write_to_attribute(eos_table_group_.id(), "independent variable names",
+                         independent_variable_names_);
+  h5::write_to_attribute(eos_table_group_.id(), "independent variable bounds",
+                         independent_variable_bounds_);
+  h5::write_to_attribute(eos_table_group_.id(), "number of points",
+                         independent_variable_number_of_points_);
+  h5::write_to_attribute(eos_table_group_.id(), "uses log spacing",
+                         independent_variable_uses_log_spacing_);
+  h5::write_to_attribute(eos_table_group_.id(), "beta equilibium",
+                         beta_equilibrium_);
+}
+
+EosTable::EosTable(const bool subfile_exists, detail::OpenGroup&& group,
+                   const hid_t /*location*/, const std::string& name)
+    : group_(std::move(group)),
+      name_(name.size() > extension().size()
+                ? (extension() == name.substr(name.size() - extension().size())
+                       ? name
+                       : name + extension())
+                : name + extension()),
+      path_(group_.group_path_with_trailing_slash() + name),
+      eos_table_group_(group_.id(), name_, h5::AccessType::ReadOnly) {
+  if (not subfile_exists) {
+    ERROR(
```

Comment:
test this

Reply:
This gets checked by the H5File class already, so just removed